### ISSUE TITLE
TEST: Run real tests on develop branch

### ISSUE THREAD

**Issue by HenrikBengtsson** [UCSF-Costello-Lab/LG3_Pipeline#6]
@ivan108, when we've now passed the ShellCheck milestone, let's run some real validation tests on it, i.e. on the `develop` branch;
```sh
$ cd /home/jocostello/shared/LG3_Pipeline
$ git checkout devel
$ git pull
```
and run your tests. When that's done, we can then merge this validated version into the master branch:
```sh
$ git checkout master
$ git merge develop
$ git push
```
and then also tag it by todays date:
```sh
$ git tag 2018-08-31
$ git push --tags
```

**Comment by ivan108** [comment]
OK, I am on it.
Just to clarify, the test I am going to run is based on real data on Patient157 (normal control and two tumors). It will take about 4 days. The run will involve only a small number of "original" scripts, lot's of other "special" scripts which were created based on original scripts to address some specific issues, will not be tested. E.g. if BQS recalibration run failed on step 3, to save time we want to restart recal from step 3 skipping steps 1 and 2, and there is a "special" script for that. Or if recal went through but for some reason germline analysis which is a part of it, failed. So we would use a "special" script to run additionally only germline analysis without recal, so that we can check relatedness..

**Comment by HenrikBengtsson** [comment]
Got it.

Could you please put down cut'n'pastable instructions here on how you run this initial 'Patient157' test?  That can become a starter for adding test scripts, e.g. `make test`, but also end-user documentation.

**Comment by ivan108** [comment]
OK,

the test is still running, meanwhile I updated develop branch with instructions and scripts for test/demo run...


BTW, is it possible to view branched code on the github?


Thanks!

________________________________

**Comment by HenrikBengtsson** [comment]
Great.

> BTW, is it possible to view branched code on the github?

Yes, just go to the 'Code' tab and there you'll find a 'Branch' button that allows you to switch to any branch.  Notice how the URL changes when you switch branch there.

**Comment by HenrikBengtsson** [comment]
I see you've relaunched 'Patient157.mut' jobs.  Did something go wrong?

**Comment by ivan108** [comment]
Yes, I forgot to link pykent directory, so one of the steps quietly failed...

**Comment by ivan108** [comment]
Just added to develop Demo_output.md doc listing expected file sizes for test/demo run...

**Comment by ivan108** [comment]
It seems only 2 first steps, trimming and alignment, are deterministic and produce exact same file sizes.
So, in general, exact file size match may be a too strict requirement...

**Comment by HenrikBengtsson** [comment]
Got it. Did you check the `md5sum` on those files too?

For results depending on a random number generator, where we won't get identical files, at least we could check for too large discrepancies in file sizes and give a warning about that.  Not sure how easy that is to do in Bash though.

**Comment by ivan108** [comment]
Just finished final step of demo run. Caught one more "quote" bug in the process. File sizes are close to the expected ones, and most importantly the final list of called mutations is identical to the original (using diff file1 file2). So, it passed "biological" test! So, yes, I think we can consider tests on develop  completed. 

As for MD5 sums, I am not sure if this is useful when most files are expected to have different sizes, which means they will have different MD5 hashes...

**Comment by HenrikBengtsson** [comment]
Great.  I've now released the "2018-09-08" version;

* I've added a NEWS file
* I've merged your Readme_demo into the README which also links to the Demo_output file
* The master branch (production) now contains the 2018-09-08 version, e.g. https://github.com/UCSF-Costello-Lab/LG3_Pipeline
* The release commit was tagged with the version making it appear on https://github.com/UCSF-Costello-Lab/LG3_Pipeline/releases

Further working, including bug fixes, should be done to the develop branch (or another branch forked from the develop branch).
